src/jit/morph.cpp

```diff
@@ -7654,6 +7663,11 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
         return result;
     }
 
+    if (((call->gtFlags & GTF_CALL_NO_RETURN) != 0) && 
+        (call->TypeGet() == TYP_VOID))
+    {
+        fgRemoveRestOfBlock = true;
+    }
```

Comment:
This was primarily caused by an assert I hit in a test. It would seem that the call node needs to have `GTF_EXCEPT` if `fgRemoveRestOfBlock` is set. I find this confusing, how come `GTF_EXCEPT` is not always set on calls to begin with? Anyway, I didn't look more into this because such cases should be rare.